src/server/json_family_test.cc

```diff
@@ -456,4 +456,37 @@ TEST_F(JsonFamilyTest, NumMultBy) {
   EXPECT_EQ(resp, R"([{"a":"a"},{"a":"a","b":2},{"a":"a","b":"b"},{"a":2,"b":"b","c":6}])");
 }
 
+TEST_F(JsonFamilyTest, Del) {
+  string json = R"(
+    {"a":{}, "b":{"a":1}, "c":{"a":1, "b":2}, "d":{"a":1, "b":2, "c":3}, "e": [1,2,3,4,5]}}
+  )";
+
+  auto resp = Run({"set", "json", json});
+  ASSERT_THAT(resp, "OK");
+
+  resp = Run({"JSON.DEL", "json", "$.d.*"});
+  EXPECT_THAT(resp, IntArg(3));
+
+  resp = Run({"GET", "json"});
+  EXPECT_EQ(resp, R"({"a":{},"b":{"a":1},"c":{"a":1,"b":2},"d":{},"e":[1,2,3,4,5]})");
+
+  resp = Run({"JSON.DEL", "json", "$.e[*]"});
+  EXPECT_THAT(resp, IntArg(5));
+
+  resp = Run({"GET", "json"});
+  EXPECT_EQ(resp, R"({"a":{},"b":{"a":1},"c":{"a":1,"b":2},"d":{},"e":[]})");
+
+  resp = Run({"JSON.DEL", "json", "$..*"});
+  EXPECT_THAT(resp, IntArg(8));
+
+  resp = Run({"GET", "json"});
+  EXPECT_EQ(resp, R"({})");
+
+  resp = Run({"JSON.DEL", "json"});
+  EXPECT_THAT(resp, IntArg(1));
+
+  resp = Run({"GET", "json"});
+  EXPECT_THAT(resp, ArgType(RespExpr::NIL));
+}
+
```

Comment:
so I found a bug in the previous iteration.
I think it justifies adding tests that cover this bug. 

also, given the increased complexity of the transformation function, I would like to see how it copes with longer expressions then just a single depth hierarchy .

can you please add tests covering transformation of `object/array/object/array` and `object/array/object` .

Reply:
In the case of the following JSON: {"a']": [1,2,3]} the key "a']" can't be described explicitly within the JSONPath due to the limitation of that syntax, I read about it in the RFC.  so I used the JSONPath "$.*" to check it can be deleted implicitly.

Reply:
looks much better!